### ISSUE TITLE
Embed velrid into avalonia panel

### ISSUE THREAD

**Issue by StefanH-AT** [mellinoe/veldrid#435]
Hi, I'm trying to integrate velrid as a 3D viewport into an avalonia ui panel. I read some other related issues and it seems like the bulk of getting this working is to set up a swapchain. Could someone explain how to set this up for avalonia? I'd highly appreciate it!

**Comment by NtFreX** [comment]
Hey, maybe it helps to check the VeldridStartup type. It creates a GraphicDevice for a SDL window.
https://github.com/mellinoe/veldrid/blob/master/src/Veldrid.StartupUtilities/VeldridStartup.cs#L101

This is the same for an android implementation:
https://github.com/mellinoe/veldrid-samples/blob/0ac284ddc8f62d8d75c8eae88cc1824c320b1caf/src/SampleBase.Android/VeldridSurfaceView.cs#L12

And here for ios:
https://github.com/mellinoe/veldrid-samples/blob/master/src/SampleBase.iOS/UIViewApplicationWindow.cs#L52

**Comment by StefanH-AT** [comment]
But how would I go about getting an SDL window from my avalonia window? I tried using the IntPtr from my avalonia window like this:
```cs
var topLevel = (TopLevel) this.GetVisualRoot();
var windowImpl = (WindowImpl) topLevel.PlatformImpl;
var handle = windowImpl.Handle.Handle;
```

then I created the graphics device like
```cs
_window = new Sdl2Window(handle, false); // Should this be true? I'm running my draw loop in a separate thread anyways to unblock avalonia
_graphicsDevice = VeldridStartup.CreateGraphicsDevice(_window, options);
```

However, the Sdl2Window doesn't seem to get properly initialized and causes an error. All values are default.

I'd appreciate some help to point me in the right direction here. I barely know anything about Window panel draw area systems or what SDL is

**Comment by zaafar** [comment]
I don't think so that NtFreX is suggesting you to use SDL2Window or try to create one.

All he is suggesting you is to see how it's done with other windows types (e.g. sdl2, android, ios) and adapt it.

**Comment by YITWEE** [comment]
Hello, are there any examples that worked using NativeControlHost?
I have tried and just can see a part of the rendered graphics when I resize windows quickly using the mouse.
Any help on this would be greatly appreciated. :)
